app/src/main/java/org/mozilla/fenix/settings/address/interactor/AddressEditorInteractor.kt

```diff
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.address.interactor
+
+import mozilla.components.concept.storage.UpdatableAddressFields
+import org.mozilla.fenix.settings.address.controller.AddressEditorController
+
+/**
+ * Interface for the address editor interactor.
+ */
+interface AddressEditorInteractor {
+
+    /**
+     * Navigates back to the autofill preference settings. Called when a user taps on the
+     * "Cancel" button.
+     */
+    fun onCancelButtonClicked()
+
+    /**
+     * Saves the provided address field into the autofill storage. Called when a user
+     * taps on the save menu item or "Save" button.
+     *
+     * @param addressFields A [UpdatableAddressFields] record to add.
+     */
+    fun onSaveAddress(addressFields: UpdatableAddressFields)
+}
+
+/**
+ * The default implementation of [AddressEditorInteractor].
+ *
+ * @param controller An instance of [AddressEditorController] which will be delegated for all
+ * user interactions.
+ */
+class DefaultAddressEditorInteractor(
+    private val controller: AddressEditorController
+) : AddressEditorInteractor {
+
+    override fun onCancelButtonClicked() {
+        controller.handleCancelButtonClicked()
+    }
+
+    override fun onSaveAddress(addressFields: UpdatableAddressFields) {
+        controller.handleSaveAddress(addressFields)
+    }
+}
```

Comment:
You aren't the first person to propose this. However, I think we would prefer to formalize such an architectural decision via a PRD and do a larger refactor on Fenix as a whole instead of having only Addresses be the one that diverges from the existing architecture.

Reply:
I added a bullet to the next weekly meeting if we wanna talk about it further then.

Reply:
I meant RFC instead of PRD. Here's some prior art https://github.com/mozilla-mobile/android-components/blob/main/docs/rfcs/0001-rfc-process.md.